public/pages/Snapshots/components/RestoreActivitiesPanel/RestoreActivitiesPanel.tsx

```diff
@@ -0,0 +1,156 @@
+    getRestoreStatus();
+  }, []);
+
+  const getRestoreStatus = async () => {
+    if (stage.indexOf("DONE") >= 0) {
+      console.log("done");
+      return;
+    }
+    try {
+      const res = await snapshotManagementService.getIndexRecovery();
+
+      if (res.ok) {
+        const response: GetIndexRecoveryResponse = res.response;
+
+        setRestoreStatus(response);
+        console.log(response);
+      } else {
+        context?.notifications.toasts.addDanger(res.error);
+      }
+    } catch (err) {
+      context?.notifications.toasts.addDanger(getErrorMessage(err, "There was a problem loading the recovery."));
+    }
+  };
+
+  const onIndexesClick = (e: React.MouseEvent) => {
+    e.preventDefault();
+    setFlyout(true);
+    console.log("index clicked");
+  };
+
+  const onCloseFlyout = () => {
+    setFlyout(false);
+  };
+
+  const setRestoreStatus = (response: GetIndexRecoveryResponse) => {
+    let minStartTime: number = 0;
+    let maxStopTime: number = 0;
+    let stageIndex: number = Infinity;
+    let doneCount: number = 0;
+    const indexes: CatSnapshotIndex[] = [];
+    const stages: string[] = ["START", "INIT", "INDEX", "FINALIZE", "DONE"];
+
+    for (let item in response) {
+      if (item.indexOf("kibana") < 0) {
+        const info = response[item as keyof GetIndexRecoveryResponse].shards[0]
+        const stage = stages.indexOf(info.stage);
+        const size = `${(info.index.size.total_in_bytes / 1024 ** 2).toFixed(2)}mb`;
+
+        const time = {
+          start_time: info.start_time_in_millis,
+          stop_time: info.stop_time_in_millis,
+        };
+
+        doneCount = stage === 4 ? doneCount + 1 : doneCount;
+        stageIndex = stage < stageIndex ? stage : stageIndex;
+        minStartTime = minStartTime && minStartTime < time.start_time ? minStartTime : time.start_time;
+        maxStopTime = maxStopTime && maxStopTime > time.stop_time ? maxStopTime : time.stop_time;
+
+        if (info.source.index) {
+          indexes.push({ index: info.source.index, "store.size": size });
+        }
+      }
+    }
+    let percent = Math.floor((doneCount / indices.length) * 100);
+    percent = stageIndex === 4 ? 100 : percent;
+
+    setStartTime(new Date(minStartTime).toLocaleString().replace(",", "  "));
+    setStopTime(new Date(maxStopTime).toLocaleString().replace(",", "  "));
+    setIndices(indexes);
+    setStage(`${stages[stageIndex]} (${percent}%)`);
+  };
+
+  const actions = [
+    <EuiButton iconType="refresh" onClick={getRestoreStatus} data-test-subj="refreshStatusButton">
+      Refresh
+    </EuiButton>,
+  ];
+
+  const indexes = `${indices.length} ${indices.length === 1 ? "Index" : "Indices"}`;
+  const restoreStatus = [
+    {
+      start_time: startTime,
+      stop_time: stopTime,
+      snapshot: snapshotId,
+      status: stage,
+      indexes: indexes,
+    },
+  ];
+  const columns = [
```

Comment:
memoize this?

Reply:
Will look into this.

Reply:
How would this be beneficial?  Am I missing something?  Let's discuss!

Reply:
done